### PR TITLE
Login Cleanup

### PR DESCRIPTION
**Pull request by coreyfloyd**
This PR is for cleaning up our login logic just to make it serviceable.

https://phabricator.wikimedia.org/T138350

After the last rushed update for new Authmanager support, we had a lot of branched code and much copy and pasting.

This PR specifically addresses:
- Sharing authentication code by creating a non-UI authentication manager
- Removed authentication logic from view controllers
- removed copy and pasted code from session singleton that supported auto login
- Removed legacy authentication code

Out of scope for this PR:
- Updating fetchers to new fetcher architecture
- Making the Authentication manager stateless (which requires the above)
- These will be addressed in T124048

### CODE REVIEW

File: Wikipedia/Code/WMFAuthenticationManager.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/**
+ *  Get a captcha for account creation. This operation will also check if the username/password is valid.
+ *
+ *  @param username The username to create
+ *  @param password The password for the new user
+ *  @param email    The email for the new user
+ *  @param captcha  The handler for the returned captcha URL
+ *  @param failure  The handler for any errors
+ */
+- (void)getAccountCreationCaptchaWithUsername:(NSString*)username password:(NSString*)password email:(nullable NSString*)email captcha:(WMFCaptchaHandler)captcha failure:(WMFErrorHandler)failure;
```
Comment (joewalsh) [inline Wikipedia/Code/WMFAuthenticationManager.h]: I did `failure: success:` for the PromiseKit changes. Should I change that to `success: failure:` for consistency or should we let it be in whatever order? I'm fine either way. I put success last due to the last param block syntax in swift.

**Comment by joewalsh** [comment]
👍

File: Wikipedia/Code/WMFAuthenticationManager.h (hunk 1 above)
Comment (coreyfloyd) [inline Wikipedia/Code/WMFAuthenticationManager.h]: @joewalsh I didn't notice that… I don't have a strong preference, you?

Comment (joewalsh) [inline Wikipedia/Code/WMFAuthenticationManager.h]: nope, no strong pref. i'm fine with it being different in different places.

**Comment by montehurd** [comment]
@coreyfloyd Logout looks to be broken.

**Comment by montehurd** [comment]
Login works. Testing acct creation now...

**Comment by montehurd** [comment]
Acct creations works nicely. Looks like just the logout fix is needed.

File: Wikipedia/Code/WMFAuthenticationManager.m [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+@property (strong, nonatomic, nullable) NSString* loginToken;
+@property (strong, nonatomic, nullable) NSString* accountCreationToken;
+
+@property (nonatomic, copy, nullable) dispatch_block_t successBlock;
+@property (nonatomic, copy, nullable) WMFCaptchaHandler captchaBlock;
+@property (nonatomic, copy, nullable) WMFErrorHandler failBlock;
+
+@end
+
+@implementation WMFAuthenticationManager
```
Comment (montehurd) [inline Wikipedia/Code/WMFAuthenticationManager.m]: This auth manager is great.

**Pull request merged**